sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/constraints/Check.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog.constraints;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.spark.SparkIllegalArgumentException;
+import org.apache.spark.sql.connector.expressions.filter.Predicate;
+
+/**
+ * A CHECK constraint.
+ * <p>
+ * A CHECK constraint defines a condition each row in a table must satisfy. Connectors can define
+ * such constraints either in SQL (Spark SQL dialect) or using a {@link Predicate predicate} if the
+ * condition can be expressed using a supported expression. A CHECK constraint can reference one or
+ * more columns. Such constraint is considered violated if its condition evaluates to {@code FALSE},
+ * but not {@code NULL}. The search condition must be deterministic and cannot contain subqueries
+ * and certain functions like aggregates or UDFs.
+ *
+ * @since 4.1.0
+ */
+public class Check extends BaseConstraint {
+
+  private final String sql;
+  private final Predicate predicate;
+
+  private Check(
+      String name,
+      String sql,
+      Predicate predicate,
+      boolean enforced,
+      ValidationStatus validationStatus,
+      boolean rely) {
+    super(name, enforced, validationStatus, rely);
+    this.sql = sql;
+    this.predicate = predicate;
+  }
+
+  /**
+   * Returns the SQL representation of the search condition (Spark SQL dialect).
+   */
+  public String sql() {
```

Comment:
Since it represents the search condition, how about predicateSQL ? or other better name. Otherwise, it looks confused.

Reply:
Agree, changed.